arangod/VocBase/LogicalCollection.cpp

```diff
@@ -1065,3 +1120,33 @@ VPackSlice LogicalCollection::keyOptions() const {
   }
   return VPackSlice(_keyOptions->data());
 }
+
+void LogicalCollection::validators(VPackBuilder& b) const {
+  VPackArrayBuilder guard(&b);
+  auto vals = std::atomic_load_explicit(&_validators, std::memory_order_relaxed);
+  for(auto const& validator : *vals) {
+    validator->toVelocyPack(b);
+  }
+}
+
+Result LogicalCollection::validate(VPackSlice s, bool validate) const {
+  if (!validate) { return {}; }
+  auto vals = std::atomic_load_explicit(&_validators, std::memory_order_relaxed);
+  for(auto const& validator : *vals) {
+    if(!validator->validate(s, VPackSlice::noneSlice(), ValidatorOperation::INSERT)) {
+      return {TRI_ERROR_BAD_PARAMETER, "validation failed: " + validator->message() };
+    }
+  }
+  return {};
+}
+
+Result LogicalCollection::validate(VPackSlice new_, VPackSlice old_, bool validate) const {
```

Comment:
I dislike the `validate` parameter here. Why do we call a validate method with a `false` parameter in case we don't want to validate? It would be simpler to just not call it.